### PR TITLE
`docker images` friendly duration gets unfriendly after a while

### LINKED ISSUE #12852: `docker images` friendly duration gets unfriendly after a while
I looked around to see if this had already been reported but I couldn't find it; sorry if it is a duplicate.

It appears that the CREATED column for `docker images` gets pretty weird once it's greater than a couple years (or maybe just greater than one year?)  Check this out:

```
$ docker images
REPOSITORY                                 TAG                 IMAGE ID            CREATED              VIRTUAL SIZE
frew/w.pl                                  latest              978d5062cf9e        14 minutes ago       727.5 MB
frew/frew-perl                             latest              ae93305f9afa        2 days ago           984 MB
frew/harbinger                             latest              ca82ab9387b2        6 days ago           767.2 MB
perl                                       latest              0679310f6c10        3 months ago         863.7 MB
perl                                       5                   0679310f6c10        3 months ago         863.7 MB
perl                                       5.20                0679310f6c10        3 months ago         863.7 MB
postgres                                   9.3                 2f007d4a5aa5        8 months ago         213.1 MB
ubuntu                                     12.04               8dbd9e392a96        2.045883 years ago   131.5 MB
base                                       ubuntu-12.10        b750fe79269d        2.097004 years ago   24.65 kB
```

Hope this helps!

### PR DESCRIPTION
**Pull request by jmzwcn**
fix #12852
Signed-off-by: Daniel Zhang jmzwcn@gmail.com

### CODE REVIEW

**Comment by jmzwcn** [comment]
How append the edited test file into this commit?

**Comment by runcom** [comment]
sorry but this is a duplicate of #12885

**Comment by RickWieman** [comment]
Nice catch, I like it!

As this is such a small changeset, can you perhaps squash the two commits (as discussed in the [contributors guide](https://github.com/docker/docker/blob/master/CONTRIBUTING.md)?

File: pkg/units/duration.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ func HumanDuration(d time.Duration) string {
 		return fmt.Sprintf("%d weeks", hours/24/7)
 	} else if hours < 24*365*2 {
 		return fmt.Sprintf("%d months", hours/24/30)
+	} else {
+		return fmt.Sprintf("%d years", hours/24/365)
```
Comment (runcom) [inline pkg/units/duration.go]: I think the else here is redunant even if it lets you access `hours` that was previously int'ed. Perhaps `return fmt.Sprintf("%d years", int(d.Hours())/24/365` would be better

**Comment by crosbymichael** [comment]
LGTM

**Comment by jmzwcn** [comment]
I have squashed the two commits.

**Comment by thaJeztah** [comment]
Thanks, LGTM

**Comment by duglin** [comment]
LGTM

**Pull request merged**